### PR TITLE
[Android] Show deprecation warnings for Android tests

### PR DESCRIPTION
**Pull request by camsim99**
This PR makes the warnings caused by the usage of deprecated APIs in the Android engine to appear when testing it.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: shell/platform/android/test_runner/build.gradle
Comment (blasten) [inline shell/platform/android/test_runner/build.gradle:21]: is it possible to turn it on by default?  Btw, how is the output different when this flag is passed?

Comment (camsim99) [reply shell/platform/android/test_runner/build.gradle:21]: Yeah, I can turn it on by default, but just FYI there are 100 deprecation warnings right now! Those print out before the tests are executed and the results are printed.

File: testing/run_tests.py
Comment (blasten) [inline testing/run_tests.py:339]: could it be set in build.gradle directly, or is the flag helpful for some situations?

File: shell/platform/android/test_runner/build.gradle
Comment (GaryQian) [reply shell/platform/android/test_runner/build.gradle]: That seems fine, if we don't turn it on, we will never migrate off of the deprecated API haha. If it prints before, we can still see test results easily at bottom of the output.

File: testing/run_tests.py
Comment (camsim99) [reply testing/run_tests.py:339]: I only added the flag in case someone wants to hide the warnings, but I can remove the flag if you don't see a need for that?

Comment (GaryQian) [inline testing/run_tests.py:339]: Might be a bit more clear if `default=True` were set in the parser.add_argument rather than depend on the value being None.

Comment (camsim99) [reply testing/run_tests.py]: @GaryQian any thoughts on this?

**Review by GaryQian — APPROVED** [review]
I'll LGTM this for now, your call on if you leave the flag in or not!

You'll also have to rebase this PR, try `git pull upstream main --rebase`, fix conflicts, then `git push origin add_deprecation_warnings -f`

**Comment by Hixie** [comment]
Could we test this by having it verify that there's no warnings on the default `flutter create` app?
And maybe even test it by verifying that there _is_ a warning on an app designed specifically to use a deprecated API?

**Comment by camsim99** [comment]
> Could we test this by having it verify that there's no warnings on the default `flutter create` app? And maybe even test it by verifying that there _is_ a warning on an app designed specifically to use a deprecated API?

This is just adding the warnings when running the Android engine tests, so we'd expect this not to affect creating/running Flutter apps at all. I just verified this behavior myself!

**Comment by Hixie** [comment]
Oh, I see. Can we make the warnings fatal then?

**Comment by Hixie** [comment]
oh, totally!

test-exempt: adding instrumentation to help fix code, warnings will be made fatal once there aren't any!

thanks!

**Pull request merged**
